src/filesystem.cc

```diff
@@ -2001,78 +2000,77 @@ class FileSystemManager {
       const std::string& file_system_prefix, const std::string& path,
       FileSystemCredential& credential);
 
-  static std::recursive_mutex
-      mu_;  // for protecting concurrent access into static variables
-  static bool is_credential_cached;
-
+  // for protecting concurrent access into static variables
+  static std::mutex mu_;
+  static bool is_credential_cached_;
   // credentials should be sorted in descending name length order
 #ifdef TRITON_ENABLE_GCS
-  static std::vector<std::pair<std::string, std::string>> gs_credential;
-  GCSFileSystem* gcs_fs;
+  static std::vector<std::pair<std::string, std::string>> gs_credential_;
+  GCSFileSystem* gcs_fs_;
 #endif  // TRITON_ENABLE_GCS
 #ifdef TRITON_ENABLE_S3
   static std::vector<std::pair<
       std::string,
       std::tuple<std::string, std::string, std::string, std::string>>>
-      s3_credential;
-  S3FileSystem* s3_fs;
+      s3_credential_;
+  S3FileSystem* s3_fs_;
 #endif  // TRITON_ENABLE_S3
 #ifdef TRITON_ENABLE_AZURE_STORAGE
   static std::vector<
       std::pair<std::string, std::pair<std::string, std::string>>>
-      as_credential;
-  ASFileSystem* as_fs;
+      as_credential_;
+  ASFileSystem* as_fs_;
 #endif  // TRITON_ENABLE_AZURE_STORAGE
-  LocalFileSystem* local_fs;
+  LocalFileSystem* local_fs_;
 };
 // initialize FileSystemManager static variables
-std::recursive_mutex FileSystemManager::mu_;
-bool FileSystemManager::is_credential_cached = false;
+std::mutex FileSystemManager::mu_;
+bool FileSystemManager::is_credential_cached_ = false;
 #ifdef TRITON_ENABLE_GCS
 std::vector<std::pair<std::string, std::string>>
-    FileSystemManager::gs_credential;
+    FileSystemManager::gs_credential_;
 #endif  // TRITON_ENABLE_GCS
 #ifdef TRITON_ENABLE_S3
 std::vector<std::pair<
     std::string,
     std::tuple<std::string, std::string, std::string, std::string>>>
-    FileSystemManager::s3_credential;
+    FileSystemManager::s3_credential_;
 #endif  // TRITON_ENABLE_S3
 #ifdef TRITON_ENABLE_AZURE_STORAGE
 std::vector<std::pair<std::string, std::pair<std::string, std::string>>>
-    FileSystemManager::as_credential;
+    FileSystemManager::as_credential_;
 #endif  // TRITON_ENABLE_AZURE_STORAGE
 
 FileSystemManager::FileSystemManager()
 {
 #ifdef TRITON_ENABLE_GCS
-  gcs_fs = NULL;
+  gcs_fs_ = NULL;
 #endif  // TRITON_ENABLE_GCS
 #ifdef TRITON_ENABLE_S3
-  s3_fs = NULL;
+  s3_fs_ = NULL;
 #endif  // TRITON_ENABLE_S3
 #ifdef TRITON_ENABLE_AZURE_STORAGE
-  as_fs = NULL;
+  as_fs_ = NULL;
 #endif  // TRITON_ENABLE_AZURE_STORAGE
-  local_fs = NULL;
+  local_fs_ = NULL;
```

Comment:
Use `nullptr`

Reply:
Switched to `unique_ptr`, no longer applicable.